services/venice-controller/src/main/java/com/linkedin/venice/controller/ProtocolVersionAutoDetectionService.java

```diff
@@ -26,11 +31,12 @@
 public class ProtocolVersionAutoDetectionService extends AbstractVeniceService {
   private static final Logger LOGGER = LogManager.getLogger(ProtocolVersionAutoDetectionService.class);
   private static final String PARENT_REGION_NAME = "parentRegion";
+  private final ScheduledExecutorService executor =
+      Executors.newSingleThreadScheduledExecutor(new DaemonThreadFactory("ProtocolVersionAutoDetectionService"));;
```

Comment:
Nice!

Reply:
Could you pass down the `LogContext` to `DaemonThreadFactory`?

Reply:
Added, please check the last commit. Thank you!